eloquentjs/6_3_groups_iterable.js

```diff
@@ -0,0 +1,94 @@
+class Group {
+    constructor() {
+        this.n = 0;
+        this.set = [];
+    }
+
+    has(item) {
+        return this.set.includes(item) ? true : false;
+    }
+
+    add(item) {
+        if (!this.has(item)) {
+            this.set.push(item);
+        } else {
+            console.log(`Set already includes item '${item}!'`);
+        }
+    }
+
+    delete(item) {
+        if (this.has(item)) {
+            this.set.splice(this.set.findIndex(j => j === item), 1);
+        } else {
+            console.log(`Set doesn't have '${item}'!`);
+        }
+    }
+
+    static from(inputArray) {
+        let output = new Group();
+
+        for (let element of inputArray) {
+            output.add(element);
+        }
+
+        return output;
+    }
+
+    [Symbol.iterator]() {
```

Comment:
Generally speaking, it's a good idea to create a separate iterator class that contains this one and return an instance of it. That way you can have multiple iterators at once and the overall interface is cleaner. Why don't you take a shot at that?